### PR TITLE
feat: updated PricedCallSpread contract

### PR DESCRIPTION
**Pull request by tgrecojs**
PR for [issue #2282](https://github.com/Agoric/agoric-sdk/issues/2282)

updated PricedCallSpread contract to take 4 digit input. After seeing the exisiting input argument had to fall between 0-100, I'm not sure if there should be a check to enforce inputs that are only 2 or 4 digits but would be happy to update if needed. 🙂

### CODE REVIEW

File: packages/zoe/src/contracts/callSpread/pricedCallSpread.js [hunk 1]
```diff
@@ -23,12 +23,15 @@ import { Position } from './position';
...
+  Number(x).toString().length < 4 ? (x * BASIS_POINTS) / PERCENT_BASE : x;
+
 /**
  * This contract implements a fully collateralized call spread. This is a
  * combination of a call option bought at one strike price and a second call
  * option sold at a higher price. The invitations are produced in pairs. The
- * creatorFacet has a method makeInvitationPair(longCollateralShare) whose
- * argument must be a number between 0 and 100. makeInvitationPair() returns two
+ * creatorFacet has a method makeInvitations(longCollateralShare) whose
+ * argument must be a number between 0 and 100. makeInvitations() returns two
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contracts/callSpread/pricedCallSpread.js:33]:
```suggestion
 * argument must be a number between 0 and 10000. makeInvitations() returns two
```

File: packages/zoe/src/contracts/callSpread/pricedCallSpread.js [hunk 2]
```diff
@@ -23,12 +23,15 @@ import { Position } from './position';
 const PERCENT_BASE = 100n;
 const BASIS_POINTS = 10000n;
 
+const evaluateInvitationInput = x =>
+  Number(x).toString().length < 4 ? (x * BASIS_POINTS) / PERCENT_BASE : x;
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contracts/callSpread/pricedCallSpread.js:26]: We shouldn't interpret the input differently depending on length.  `2500` means 2500 basis points, and `50` should mean 50 basis points.

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]
Thanks for this update!

It would be a mistake to treat 2-digit inputs as percent, and 4-digit inputs as basis points. Instead, we should assert that the input is `<= 10000` (no reason to convert to string for this check), and then always treat it as basis points.

The first failing CI test was because we have a requirement that individual commits have to conform to ["conventional commits"]( https://www.conventionalcommits.org ). You'll have to reword the commit to get this to pass.

The second failing test was because the pre-checks also ensure consistency with our docs, and the docs have tests on their sample code to ensure it doesn't become out-of-date with respect to the codebase. We won't be able to submit this change until we have an update for the docs repo as well. I can help with that if we don't have this well enough documented. Once there's a PR for the docs repo, the wiki has some instructions for [tying the two PRs together](https://github.com/Agoric/agoric-sdk/wiki/Cross-repo-Integration-Tests) so the tests will pass.

File: packages/zoe/src/contracts/callSpread/pricedCallSpread.js (hunk 2 above)
Comment (tgrecojs) [reply packages/zoe/src/contracts/callSpread/pricedCallSpread.js:26]: 👍

File: packages/zoe/src/contracts/callSpread/pricedCallSpread.js [hunk 3]
```diff
@@ -154,14 +157,9 @@ const start = zcf => {
     });
   }
 
-  // TODO(2282): change the API so the caller can provide share in basis points
-  //  rather than percent
-  function makeInvitationPair(longCollateralShare) {
-    const longPercent = makeRatio(
```
Comment (tgrecojs) [inline packages/zoe/src/contracts/callSpread/pricedCallSpread.js:156]: @Chris-Hibbert am I on the right track here 🤔 or am I missing it?

```js
const longValue = longcollateralShare <= 10000 ? 
longcollateralShare 
: 
makeRatio(longCollateralShare * BASIS_POINTS) / PERCENT_BASE,
      brands.Collateral,
      BASIS_POINTS,
    );

 const longInvitation = makeOptionInvitation(Position.LONG, longValue);
```

**Comment by tgrecojs** [comment]
> Thanks for this update!
> 
> It would be a mistake to treat 2-digit inputs as percent, and 4-digit inputs as basis points. Instead, we should assert that the input is `<= 10000` (no reason to convert to string for this check), and then always treat it as basis points.
> 

No problem at all. Happy to help! 🙂 

I added a comment about a possible solution. hoping I'm on track 🤞 but let me know if I'm still misunderstanding. 

> The first failing CI test was because we have a requirement that individual commits have to conform to ["conventional commits"](https://www.conventionalcommits.org). You'll have to reword the commit to get this to pass.
> 
> The second failing test was because the pre-checks also ensure consistency with our docs, and the docs have tests on their sample code to ensure it doesn't become out-of-date with respect to the codebase. We won't be able to submit this change until we have an update for the docs repo as well. I can help with that if we don't have this well enough documented. Once there's a PR for the docs repo, the wiki has some instructions for [tying the two PRs together](https://github.com/Agoric/agoric-sdk/wiki/Cross-repo-Integration-Tests) so the tests will pass.

Thanks for this input as well. I did try to implement the conventional commit style but always tend to muff up, especially within repos I'm not regularly working in 😬 . As for the docs, I can help out on that end as well. Going to look into that cross-repo wiki page you linked to above.

File: packages/zoe/src/contracts/callSpread/pricedCallSpread.js (hunk 3 above)
Comment (Chris-Hibbert) [reply packages/zoe/src/contracts/callSpread/pricedCallSpread.js:156]: Oops!  I entered this comment a while ago, but apparently never submitted. Sorry about that.

**Review by Chris-Hibbert — COMMENTED** [review]
Sorry, I thought I had sent this, but I see that it's still waiting for my action.

**Comment by ivanlei** [comment]
This PR has gone stale and is being closed. Please reopen if there is a desire to continue the review process.

**Pull request closed without merging**